svg.go

```diff
@@ -146,50 +145,54 @@ func MustSVGFromReader(r io.Reader) *SVG {
 // small subset of an SVG currently. Specifically, the "viewBox" attribute and any "d" attributes from enclosed SVG
 // "path" elements.
 func NewSVGFromReader(r io.Reader) (*SVG, error) {
-	var svgXML struct {
-		ViewBox string `xml:"viewBox,attr"`
-		Paths   []struct {
-			Path string `xml:"d,attr"`
-		} `xml:"path"`
-	}
-	if err := xml.NewDecoder(r).Decode(&svgXML); err != nil {
-		return nil, errs.NewWithCause("unable to decode SVG", err)
-	}
-	svg := &SVG{scaledPathMap: make(map[Size]*Path)}
-	var width, height string
-	if parts := strings.Split(svgXML.ViewBox, " "); len(parts) == 4 {
-		width = parts[2]
-		height = parts[3]
-	}
-	v, err := strconv.ParseFloat(width, 64)
-	if err != nil || v < 1 || v > 4096 {
-		return nil, errs.NewWithCause("unable to determine SVG width", err)
+	svg, err := svg.Parse(r, svg.StrictErrorMode)
+	if err != nil {
+		return nil, err
 	}
-	svg.size.Width = float32(v)
-	v, err = strconv.ParseFloat(height, 64)
-	if err != nil || v < 1 || v > 4096 {
-		return nil, errs.NewWithCause("unable to determine SVG height", err)
+
+	s := &SVG{
+		paths:               make([]*Path, len(svg.SvgPaths)),
+		scaledCombinedPaths: make(map[Size]*Path),
+		size: Size{
+			Width:  float32(svg.ViewBox.W),
+			Height: float32(svg.ViewBox.H),
+		},
 	}
-	svg.size.Height = float32(v)
-	for i, svgPath := range svgXML.Paths {
-		var p *Path
-		if p, err = NewPathFromSVGString(svgPath.Path); err != nil {
-			return nil, errs.NewWithCausef(err, "unable to decode SVG: path element #%d", i)
-		}
-		if svg.unscaledPath == nil {
-			svg.unscaledPath = p
-		} else {
-			svg.unscaledPath.Path(p, false)
+
+	for i, path := range svg.SvgPaths {
+		p, err := newPathFromSvgPath(path)
+		if err != nil {
+			return nil, err
 		}
+		s.paths[i] = p
 	}
-	return svg, nil
+
+	return s, nil
 }
 
 // Size returns the original size.
 func (s *SVG) Size() Size {
 	return s.size
 }
 
+// CombinedPath combines all paths for an SVG in to a single path,
+// by extending the first path will all of the other paths.
+// The combined path will have the fill and stroke attributes of the first path.
+func (s *SVG) CombinedPath() *Path {
```

Comment:
I know I asked you to restore the `PathScaledTo()` and `PathForSize()` methods, but I'm reconsidering that now after looking at why & how these are being used in my other projects. The goal was really just to be able to draw a scaled SVG image where needed. That can probably be accomplished in the `DrawInRect()` method... it's possible the existing code already works fine and didn't need those extra steps, but was kept without checking if they could be removed.

Reply:
In order to move this PR forward, and not introduce breaking changes, I suggest that the two functions are marked as deprecated. And you can follow up with their eventual removal later.